scripts/generate/templates/Packing.js

```diff
@@ -0,0 +1,63 @@
+const format = require('../format-lines');
+const { product } = require('../../helpers');
+const { SIZES } = require('./Packing.opts');
+
+// TEMPLATE
+const header = `\
+pragma solidity ^0.8.20;
+
+/**
+ * @dev Helper library packing and unpacking multiple values into bytesXX.
+ */
+// solhint-disable func-name-mixedcase
+`;
+
+const errors = `\
+  error OutOfRangeAccess();
+`;
+
+const pack = (left, right) => `\
+  function pack_${left}_${right}(bytes${left} left, bytes${right} right) internal pure returns (bytes${
+    left + right
+  } result) {
+    assembly ("memory-safe") {
+      result := or(left, shr(${8 * left}, right))
+    }
+  }
+`;
+
+const extract = (outer, inner) => `\
+  function extract_${outer}_${inner}(bytes${outer} self, uint8 offset) internal pure returns (bytes${inner} result) {
+    if (offset > ${outer - inner}) revert OutOfRangeAccess();
+    assembly ("memory-safe") {
+      result := and(shl(mul(8, offset), self), shl(${256 - 8 * inner}, not(0)))
+    }
+  }
+`;
+
+const replace = (outer, inner) => `\
+  function replace_${outer}_${inner}(
+    bytes${outer} self,
+    bytes${inner} value,
+    uint8 offset
+  ) internal pure returns (bytes${outer} result) {
+    bytes${inner} oldValue = extract_${outer}_${inner}(self, offset);
+    assembly ("memory-safe") {
+      result := xor(self, shr(mul(8, offset), xor(oldValue, value)))
+    }
+  }
+`;
```

Comment:
I would avoid the underscore if possible.

How do you feel using `L${left}R${right}` to represent left and right? Alternatively, I was thinking of `b${left}b${right}` to symbolize they're `bytes` but didn't get convinced.

```suggestion
const pack = (left, right) => `\
  function packL${left}R${right}(bytes${left} left, bytes${right} right) internal pure returns (bytes${
    left + right
  } result) {
    assembly ("memory-safe") {
      result := or(left, shr(${8 * left}, right))
    }
  }
`;

const extract = (outer, inner) => `\
  function extractL${outer}R${inner}(bytes${outer} self, uint8 offset) internal pure returns (bytes${inner} result) {
    if (offset > ${outer - inner}) revert OutOfRangeAccess();
    assembly ("memory-safe") {
      result := and(shl(mul(8, offset), self), shl(${256 - 8 * inner}, not(0)))
    }
  }
`;

const replace = (outer, inner) => `\
  function replaceL${outer}R${inner}(
    bytes${outer} self,
    bytes${inner} value,
    uint8 offset
  ) internal pure returns (bytes${outer} result) {
    bytes${inner} oldValue = extract_${outer}_${inner}(self, offset);
    assembly ("memory-safe") {
      result := xor(self, shr(mul(8, offset), xor(oldValue, value)))
    }
  }
`;
```

Reply:
In the case of `extract` and `replace`, they're not really "left" and "right" arguments. But some other letters could work.

Reply:
U and L for upper and lower might be another option

Reply:
IMO, using L/R or U/L is confusing. I think that `_` is clearer.

Sure, it not mixedcase, but I feel `pack_8_8(x, y)` is way more obvious than `packL8R8(x,y)`

Reply:
Right, I agree is clearer, I just don't trigger everyone's `solhint` rules when they update to 5.1 and don't have a better option 😢 

I'm solving in the lack of alternatives